src/qasm-simulator-cpp/src/engines/vector_engine.hpp

```diff
@@ -172,10 +172,13 @@ void VectorEngine::execute(const Circuit &prog, BaseBackend<QubitVector> *be,
   // Check to see if circuit is ideal and allows for measurement optimization
   if (prog.opt_meas && prog.noise.ideal) {
 
+    // This optimization replaces the shots by a single shot + sampling
```

Comment:
Could you add a second comment line here: `// We need to subtract the additional shots added by BaseEngine class`